lib/http/importComponents.ts

```diff
@@ -0,0 +1,59 @@
+import { createApiClient } from "../api";
+import FormData from "form-data";
+import fs from "fs";
+
+export interface ImportComponentResponse {
+  componentsInserted: number;
+  firstImportedId: string;
+}
+
+export interface CsvColumnMapping {
+  text: number;
+  name: string;
+  notes?: number;
+  tags?: number;
+  status?: number;
+  componentId?: number;
+}
+
+export async function importComponents(
+  path: string,
+  options: {
+    csvColumnMapping?: CsvColumnMapping;
+  }
+): Promise<ImportComponentResponse> {
+  const api = createApiClient();
+
+  const form = new FormData();
+  form.append("import", fs.createReadStream(path));
+
+  const requestOptions = {
+    method: "POST",
+    url: "/v1/components/file",
+    params: {
+      ...(options.csvColumnMapping?.name
+        ? { name: `[${options.csvColumnMapping.name}]` }
+        : {}),
+      ...(options.csvColumnMapping?.text
+        ? { text: options.csvColumnMapping.text }
+        : {}),
+    },
+    headers: {
+      "content-type": "multipart/form-data",
+    },
+    data: form,
+  };
+
+  try {
+    const { data } = await api(requestOptions);
+
+    return data;
+  } catch (e: any) {
+    console.error("Failed to import file.");
+    console.error(e.response?.data?.message || e.message);
```

Comment:
Maybe we should parse `error.response.body.code` (assuming that this catch can be triggered by `ApiError`s being thrown on the backend) and have user-friendly copy that is shown rather than the raw error?

Reply:
`e.response?.data?.message` should already be user-friendly copy (at least that's what I've been doing for API errors). `e.message` is a safety net that will just print the code if no user-friendly message is sent from the API. I'm thinking we should make updates to the backend if something is showing up as non-user-friendly, but do you think we need custom messages for the CLI?

Reply:
Ahhhhh you're totally right I forgot about that!